src/odemis/model/_dataflow.py

```diff
@@ -242,26 +242,30 @@ def _update_pipe_hwm(self):
         """
         updates the high water mark option of OMQ pipe according to max_discard
         """
-        if (self._max_discard == self._max_discard_last_update
-            or self.pipe is None
-           ):
-            return
-
-        # When discarding, allow a bit of delay, but nothing more: if more than
-        # 2 DataArray (= 2*3 msg) already queued, the _newest_ one will be dropped.
-        # The best would be to drop the _oldest_ messages.
-        # Note: before ZMQ 4.2, it was necessary to unbind/bind the pipe to update the HWM, however,
-        # this automatically drops all subscribers. And sometimes, even if there are no subscribers,
-        # it crashes ZMQ. From ZMQ 4.2, it's possible to change the HWM without rebinding.
-        hwm = 6 if self._max_discard else 3000
-        self.pipe.sndhwm = hwm
-        self._max_discard_last_update = self._max_discard
-        logging.debug("Updating 0MQ HWM to %d on %s (max_discard = %s)",
-                       hwm, self._global_name, self._max_discard)
-
-        # TODO: in ZMQ v4, ZMQ_CONFLATE allows to have a queue of 1 message
-        # containing only the newest message. That sounds closer to what we
-        # need (though, currently multi-part messages are not supported).
+        # As of 0MQ v4.3.4, changing the HWM while connected still doesn't work correctly. It can
+        # crash the server, or just drop messages unexpectedly. So, we just leave to a large
+        # number and let the client drop the messages.
```

Comment:
can you update the comment to explain what the code below the `return` is for and when it can be uncommented?